test/new-e2e/tests/cspm/cspm_test.go

```diff
@@ -0,0 +1,296 @@
+	})
+	assert.NoError(s.T(), err)
+	assert.Len(s.T(), res.Items, 1)
+	agentPod := res.Items[0]
+	_, _, err = s.Env().KubernetesCluster.KubernetesClient.PodExec("datadog", agentPod.Name, "security-agent", []string{"security-agent", "compliance", "check", "--dump-reports", "/tmp/reports", "--report"})
+	assert.NoError(s.T(), err)
+	dumpContent, _, err := s.Env().KubernetesCluster.KubernetesClient.PodExec("datadog", agentPod.Name, "security-agent", []string{"cat", "/tmp/reports"})
+	assert.NoError(s.T(), err)
+	findings, err := parseFindingOutput(dumpContent)
+	assert.NoError(s.T(), err)
+	s.checkFindings(findings, mergeFindings(expectedFindingsMasterEtcdNode, expectedFindingsWorkerNode))
+}
+
+func (s *cspmTestSuite) TestMetrics() {
+	s.T().Log("Waiting for datadog.security_agent.compliance.running metrics")
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+
+		metrics, err := s.Env().FakeIntake.Client().FilterMetrics("datadog.security_agent.compliance.running")
+		assert.NoError(c, err)
+		assert.NotEmpty(c, metrics)
+		s.T().Log("Metrics found: datadog.security_agent.compliance.running")
+	}, 2*time.Minute, 10*time.Second)
+
+	s.T().Log("Waiting for datadog.security_agent.compliance.containers_running metrics")
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		metrics, err := s.Env().FakeIntake.Client().FilterMetrics("datadog.security_agent.compliance.containers_running")
+		assert.NoError(c, err)
+		assert.NotEmpty(c, metrics)
+		s.T().Log("Metrics found: datadog.security_agent.compliance.containers_running")
+	}, 2*time.Minute, 10*time.Second)
+
+}
+func (s *cspmTestSuite) checkFindings(findings, expectedFindings findings) {
+	s.T().Helper()
+	checkedRule := []string{}
+	for expectedRule, expectedRuleFindinds := range expectedFindings {
+		assert.Contains(s.T(), findings, expectedRule)
+		for _, expectedFinding := range expectedRuleFindinds {
+			found := false
+			for _, finding := range findings[expectedRule] {
+				if isSubset(expectedFinding, finding) {
+					found = true
+					break
+				}
+			}
+			assert.Truef(s.T(), found, "unexpected finding %v  for rule %s", findings[expectedRule], expectedRule)
+			checkedRule = append(checkedRule, expectedRule)
+		}
+	}
+	for rule, ruleFindings := range findings {
+		if slices.Contains(checkedRule, rule) {
+			continue
+		}
+		for _, ruleFinding := range ruleFindings {
+			fmt.Printf("rule %s finding %v\n", rule, ruleFinding["result"])
+		}
+	}
+	for rule, ruleFindings := range findings {
+		if slices.Contains(checkedRule, rule) {
+			continue
+		}
+		for _, ruleFinding := range ruleFindings {
+			assert.NotContains(s.T(), []string{"failed", "error"}, ruleFinding["result"], fmt.Sprintf("finding for rule %s not expected to be in failed or error state", rule))
+		}
+	}
+
+}
+
+func isSubset(a, b map[string]string) bool {
+	for k, v := range a {
+		if b[k] != v {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
We need to differentiate between an element which is absent from the map and an element which is present with an empty value.

The current implementation would return `true` for `isSubset(map[string]string{"foo":""}, map[string]string{})` which is weird because a non-empty map cannot be a subset of the empty map.

```suggestion
func isSubset(a, b map[string]string) bool {
	for k, v := range a {
		if vb, found := b[k]; !found || vb != v {
			return false
		}
	}
	return true
}
```

Reply:
Oh yeah, did not think about that edge case and Go defaulting non-present key to default value for the type. Applied your suggestion!